frontend/src/modules/WellCompletion/settings.tsx

```diff
@@ -0,0 +1,357 @@
+                setSelectedEnsembleIdent(fixedEnsembleIdent);
+            }
+        },
+        [ensembleSet, selectedEnsembleIdent]
+    );
+
+    React.useEffect(() => {
+        // Update plot when new query data from Sumo is retrieved
+        if (wellCompletionQuery.data) {
+            if (!wellCompletionsDataAccessor.current) {
+                wellCompletionsDataAccessor.current = new WellCompletionsDataAccessor();
+            }
+
+            wellCompletionsDataAccessor.current.parseWellCompletionsData(wellCompletionQuery.data);
+
+            wellCompletionsDataAccessor.current.setSearchWellText(searchWellText);
+            wellCompletionsDataAccessor.current.setHideZeroCompletions(hideZeroCompletions);
+
+            // Update available time steps
+            const allTimeSteps = wellCompletionsDataAccessor.current.getTimeSteps();
+            if (availableTimeSteps != allTimeSteps) {
+                setAvailableTimeSteps(allTimeSteps);
+            }
+
+            // Update selected time step indices if not among available time steps
+            let timeStepIndex = selectedTimeStepOptions.timeStepIndex;
+            if (typeof timeStepIndex === "number" && allTimeSteps.length < timeStepIndex) {
+                timeStepIndex = 0;
+            }
+            if (Array.isArray(timeStepIndex)) {
+                let [startIndex, endIndex] = timeStepIndex;
+                if (startIndex > allTimeSteps.length) {
+                    startIndex = allTimeSteps.length - 1;
+                }
+                if (endIndex > allTimeSteps.length) {
+                    endIndex = allTimeSteps.length - 1;
+                }
+                timeStepIndex = [startIndex, endIndex];
+            }
+            if (timeStepIndex !== selectedTimeStepOptions.timeStepIndex) {
+                setSelectedTimeStepOptions((prev) => ({
+                    timeStepIndex,
+                    timeAggregation: prev.timeAggregation,
+                }));
+            }
+
+            if (!allTimeSteps || timeStepIndex === null || !selectedTimeStepOptions.timeAggregation) {
+                setPlotData(null);
+                return;
+            }
+            createAndSetPlotData(allTimeSteps, timeStepIndex, selectedTimeStepOptions.timeAggregation);
+        }
+    }, [wellCompletionQuery.data]);
+
+    React.useEffect(() => {
+        // Update plot when settings are adjusted, but no new query data is retrieved
+        if (
+            !wellCompletionsDataAccessor.current ||
+            !availableTimeSteps ||
+            selectedTimeStepOptions.timeStepIndex === null ||
+            !selectedTimeStepOptions.timeAggregation
+        ) {
+            return;
+        }
+
+        createAndSetPlotData(
+            availableTimeSteps,
+            selectedTimeStepOptions.timeStepIndex,
+            selectedTimeStepOptions.timeAggregation
+        );
+    }, [selectedTimeStepOptions, searchWellText, hideZeroCompletions]);
```

Comment:
Consider moving this from effect to handle functions. Otherwise, incomplete deps array.